app/services/footnote_finder_service.rb

```diff
@@ -0,0 +1,137 @@
+# This service executes 5 queries to search for footnotes
+#
+# The first query searches for matching footnotes based on their description
+#
+# This query result is used to inform the second and third query which searches
+# for footnotes based on their associations to goods nomenclatures and measures
+#
+# These are grouped and combined by footnote type and id
+#
+# Finally footnotes are retrieved by their id and type and returned and associated with there active goods nomenclatures
+class FootnoteFinderService
+  def initialize(type, id, description)
+    @type = type
+    @id = id
+    @description = description
+  end
+
+  def call
+    return [] if description_types_and_ids.empty? && type.blank? && id.blank?
+    return [] if goods_nomenclature_sids.empty?
+
+    Api::V2::FootnoteSearch::FootnotePresenter.wrap(
+      footnotes,
+      grouped_goods_nomenclatures,
+    )
+  end
+
+  private
+
+  attr_reader :type, :id, :description
+
+  def footnotes
+    all_ids_and_types = all_matching_goods_nomenclatures.keys.map do |key|
+      type = key[0..1]
+      id = key[2..]
+
+      [type, id]
+    end
+
+    Footnote
+      .actual
+      .with_footnote_types_and_ids(all_ids_and_types)
+      .eager(:footnote_descriptions)
+      .all
+  end
+
+  def grouped_goods_nomenclatures_by_goods_nomenclature_association
+    GoodsNomenclature
+      .actual
+      .with_footnotes
+      .with_footnote_type_id(type)
+      .with_footnote_id(id)
+      .with_footnote_types_and_ids(description_types_and_ids)
+      .distinct(
+        :footnotes__footnote_type_id,
+        :footnotes__footnote_id,
+        :goods_nomenclatures__goods_nomenclature_item_id,
+        :goods_nomenclatures__producline_suffix,
+      )
+      .order(
+        :footnotes__footnote_type_id,
+        :footnotes__footnote_id,
+        :goods_nomenclatures__goods_nomenclature_item_id,
+        :goods_nomenclatures__producline_suffix,
+      )
+      .select_append(:footnotes__footnote_type_id, :footnotes__footnote_id)
+      .all
+      .group_by { |goods_nomenclature| "#{goods_nomenclature[:footnote_type_id]}#{goods_nomenclature[:footnote_id]}" }
+      .transform_values do |goods_nomenclatures|
+        goods_nomenclatures.map { |goods_nomenclature| goods_nomenclature[:goods_nomenclature_sid] }
+      end
+  end
+
+  def grouped_goods_nomenclatures_by_measure_association
+    Measure
+      .with_regulation_dates_query
+      .actual
+      .with_footnotes
+      .with_footnote_type_id(type)
+      .with_footnote_id(id)
+      .with_footnote_types_and_ids(description_types_and_ids)
```

Comment:
some of these method named "with_..." are join,
others are where, it could be a bit confusing:

for example:
`
.with_footnotes  # this one is a join
.with_footnote_type_id(type) # this one is a where
`
maybe we should keep the `join` naming?
for example:
```      
.with_footnotes # -> join_footnotes
```
or maybe call 
` .with_footnote_id(id)`

` .select_footnote_id(id)`